opslevel/helpers.go

```diff
@@ -0,0 +1,16 @@
+package opslevel
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+)
+
+type TerraformSource interface {
+	Get(ctx context.Context, target interface{}) diag.Diagnostics
+}
+
+func read[T any](ctx context.Context, d *diag.Diagnostics, state TerraformSource) T {
+	var data T
+	d.Append(state.Get(ctx, &data)...)
+	return data
+}
```

Comment:
This was added as a way to easily reduce boiler plate in each Create, Read, Update and Delete function.  The pattern looks really good so as a follow up i would add this to every resource.